test/Microsoft.NET.Build.Tests/GivenThatWeWantToVerifyNuGetReferenceCompat.cs

```diff
@@ -180,6 +141,140 @@ public class GivenThatWeWantToVerifyNuGetReferenceCompat : SdkTest, IClassFixtur
+                //  See https://github.com/dotnet/sdk/issues/1077
+                return;
+            }
+
+            var testProjectName = targetFramework.Replace(".", "_") + "implicit_ptf";
+
+            var testProjectTestAsset = CreateTestAsset(testProjectName, targetFramework);
+
+            var restoreCommand = testProjectTestAsset.GetRestoreCommand(relativePath: testProjectName);
+            restoreCommand.AddSource(Path.GetDirectoryName(_net461PackageReference.NupkgPath));
+            restoreCommand.Execute().Should().Pass();
+
+            var buildCommand = new BuildCommand(
+                Stage0MSBuild,
+                Path.Combine(testProjectTestAsset.TestRoot, testProjectName));
+            buildCommand.Execute().Should().Pass();
+        }
+
+        [WindowsOnlyTheoryAttribute]
+        [InlineData("netstandard1.6")]
+        [InlineData("netcoreapp1.1")]
+        public void Net461_is_not_implicit_for_Netstandard_and_Netcore_less_than_20(string targetFramework)
+        {
+            var testProjectName = targetFramework.Replace(".", "_") + "non_implicit_ptf";
+
+            var testProjectTestAsset = CreateTestAsset(testProjectName, targetFramework);
+
+            var restoreCommand = testProjectTestAsset.GetRestoreCommand(relativePath: testProjectName);
+            restoreCommand.AddSource(Path.GetDirectoryName(_net461PackageReference.NupkgPath));
+            restoreCommand.Execute().Should().Fail();
+        }
+
+        [WindowsOnlyFact]
+        public void It_is_possible_to_disabled_net461_implicit_package_target_fallback()
+        {
+            if (UsingFullFrameworkMSBuild)
+            {
+                //  Fullframework NuGet versioning on Jenkins infrastructure issue
+                //        https://github.com/dotnet/sdk/issues/1041
+
+                //  Disabled on full framework MSBuild until CI machines have VS with bundled .NET Core / .NET Standard versions
+                //  See https://github.com/dotnet/sdk/issues/1077
+                return;
+            }
+
+            const string testProjectName = "netstandard20_disabled_ptf";
+
+            var testProjectTestAsset = CreateTestAsset(
+                testProjectName,
+                "netstandard2.0",
+                new Dictionary<string, string> { {"DisableImplicitPackageTargetFallback", "true" } });
+
+            var restoreCommand = testProjectTestAsset.GetRestoreCommand(relativePath: testProjectName);
+            restoreCommand.AddSource(Path.GetDirectoryName(_net461PackageReference.NupkgPath));
+            restoreCommand.Execute().Should().Fail();
+        }
+
+        private TestAsset CreateTestAsset(
+            string testProjectName,
+            string targetFramework,
+            Dictionary<string, string> additionalProperties = null)
+        {
+            _net461PackageReference = CreateNet461Package();
```

Comment:
It feels messy to use a field of the class to pass this object around.  I think I'm used to tests not storing any instance state.